src/test/Utils.sol

```diff
@@ -1,67 +1,210 @@
+    struct LaunchParams {
+        address template;
+        string name;
+        string stakingTokenUrl;
+        uint64 startTime;
+        address vaultFactory;
+        address[] bonusTokens;
+        address owner;
+        bytes initParams;
+    }
+
+    string public constant LOCK_EVENT = "Lock";
+    string public constant UNLOCK_EVENT = "Unlock";
+
+    function vm() public returns (Vm) {
+        return Vm(address(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D));
+    }
+
+    function launchProgram(
+        AludelFactory factory,
+        LaunchParams memory params
+    ) internal returns(address program) {
+        program = factory.launch(
+        params.template,
+            params.name,
+            params.stakingTokenUrl,
+            params.startTime,
+            params.vaultFactory,
+            params.bonusTokens,
+            params.owner,
+            params.initParams
+        );
+    }
+
+    function createInstance(address owner, CrucibleFactory crucibleFactory) internal returns (Crucible crucible) {
+        vm().prank(owner);
+        return Crucible(payable(crucibleFactory.create("")));
+    }
+
+    function createInstance(User user, CrucibleFactory crucibleFactory) internal returns (Crucible crucible) {
+        vm().prank(user.addr());
+        return Crucible(payable(crucibleFactory.create("")));
+    }
+
+    function getLockPermission(
+        User signer,
+        IUniversalVault crucible,
+        IAludel delegate,
+        ERC20 token,
+        uint256 amount
+    ) internal returns (bytes memory) {
+        return getPermission(
+            signer,
+    		LOCK_EVENT,
+    		address(crucible),
+    		address(delegate),
+    		address(token),
+    		amount,
+    		crucible.getNonce()
+        );
+    }
+
+    function getUnlockPermission(
+        User signer,
+        IUniversalVault crucible,
+        IAludel delegate,
+        ERC20 token,
+        uint256 amount
+    ) internal returns (bytes memory) {
+        return getPermission(
+    		signer,
+    		UNLOCK_EVENT,
+    		address(crucible),
+    		address(delegate),
+    		address(token),
+    		amount,
+    		crucible.getNonce()
+        );
+    }
+
+    function stake(
+        User staker,
+        IUniversalVault crucible,
+        IAludel aludel,
+        ERC20 token,
+        uint256 amount
+    ) internal {
+        bytes memory lockSig = getLockPermission(
+            staker, crucible, aludel, token, amount
+        );
+
+        aludel.stake(address(crucible), amount, lockSig);
+    }
+
+    function unstake(
+        User staker,
+        IUniversalVault crucible,
+        IAludel aludel,
+        ERC20 token,
+        uint256 amount
+    ) internal {
+        bytes memory unlockSig = getUnlockPermission(
+            staker, crucible, aludel, token, amount
+        );
+
+        aludel.unstakeAndClaim(address(crucible), amount, unlockSig);
+    }
+
+    function fundMockToken(
```

Comment:
this function is unused

Reply:
It's only for the sake of sanity.

Make some lines short because removes the need to cast the ERC20 token to address.
It's used in #40 tests